Getting-Started-With-Orleans/Grain-Persistence.md

```diff
@@ -297,4 +325,35 @@ Any other failure conditions from a write operation _should_ cause the write `Ta
+The basic storage provider for Azure Table encodes state data fields into a single table column using Orleans binary serialization.
+
+
+## ADO.NET Persistence Rationale <a name="ADONETPersistenceRationale"></a>
+
+The principles for ADO.NET backed persistence storage are:
+
+1. Keep business critical data safe an accessible while data, the format of data and code evolve.
+2. Take advantenge of vendor and storage specific functionality.
+
+In practice this means adhering to [ADO.NET implementation goals](../Runtime-Implementation-Details/Relational-Storage)
+and some added twists in ADO.NET specific storage provider that allow evolving the shape of the data in the storage.
+
+In addition to the usual storage provider capabilities, the ADO.NET provider has built-in capability to
+
+1. Change storage data format from one format to another format (e.g. from JSON to binary) when roundtripping state.
+2. Shape the type to be saved or read from the storage in arbitrary ways. This helps to evolve the version state.
+3. Stream data out of the database.
+
+Both `1.` and `2.` can be applied on arbitrary decision parameters, such as *grain ID*, *grain type*, *payload data*.
+
+This happen so that one chooses a format, e.g. [Simple Binary Encoding (SBE)](https://github.com/real-logic/simple-binary-encoding) and implements
+(IStorageDeserializer)[https://github.com/dotnet/orleans/blob/master/src/OrleansSQLUtils/Storage/Provider/IStorageDeserializer.cs] and [IStorageSerializer](https://github.com/dotnet/orleans/blob/master/src/OrleansSQLUtils/Storage/Provider/IStorageSerializer.cs).
+The built-in (de)serializers have been built using this method. The [OrleansStorageDefault<format>(De)Serializer](https://github.com/dotnet/orleans/tree/master/src/OrleansSQLUtils/Storage/Provider) can be used as examples
+on how to implement other formats.
+
+When the (de)serializers have been implemented, they need to ba added to the `StorageSerializationPicker` property in [AdoNetStorageProvider](https://github.com/dotnet/orleans/blob/master/src/OrleansSQLUtils/Storage/Provider/AdoNetStorageProvider.cs).
+This is an implementation of [IStorageSerializationPicker](https://github.com/dotnet/orleans/blob/master/src/OrleansSQLUtils/Storage/Provider/IStorageSerializationPicker.cs). By default
+[StorageSerializationPicker](https://github.com/dotnet/orleans/blob/master/src/OrleansSQLUtils/Storage/Provider/StorageSerializationPicker.cs) will be used. And example of changing data storage format
+or using (de)serializers can be seen at [RelationalStorageTests]https://github.com/dotnet/orleans/blob/master/test/TesterInternal/StorageTests/Relational/RelationalStorageTests.cs).
+
+Unfortunately currently there is no method to expose this to Orleans application consumption as there there is no method to access the framework created [AdoNetStorageProvider](https://github.com/dotnet/orleans/blob/master/src/OrleansSQLUtils/Storage/Provider/AdoNetStorageProvider.cs) instance.
```

Comment:
Or other way, feels OK to tell these features are there here then and start an issue/PR about exposing. Would you regard it an OK angle to this? Even then I should revise the language, remove "Unfortunately", I mean. I'm heading soon off to Gitter to ask if someone knows a way to have a reference to the concrete provider either when instantiating or say, in Bootstrapper.